jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -4569,6 +4593,21 @@ public String(char[] data, int start, int length) {
 		}
 	}
 
+	String(char[] data, int length, boolean compressed) {
```

Comment:
I don't think there is a need for this constructor to exist. Can we not just reuse the one below with `length == 0`? These things would typically get inlined and the constant will be propagated thus ending up with the same implementation as you have here.